537305 Create Home Screen Skeleton & Tiles

venkat-eviden opened the pull request.

stephan-williamson commented on src/Epr.Reprocessor.Exporter.UI/Controllers/HomeController.cs:
```diff
@@ -1,20 +1,37 @@
     {
-        return View();
+        var userData = User.GetUserData();
+        //var organisation = userData.Organisations.First(x => x.OrganisationRole == OrganisationRoles.Producer);
```
Remove commented line

stephan-williamson commented on src/Epr.Reprocessor.Exporter.UI/Program.cs:
```diff
@@ -104,11 +109,7 @@
-//TODO: Check if JourneyAccessCheckerMiddleware required
-//app.UseMiddleware<JourneyAccessCheckerMiddleware>();
-//TODO: Check if data AnalyticsCookieMiddleware required
+app.UseMiddleware<UserDataCheckerMiddleware>();
```
Duplicate app.UseMiddleware<UserDataCheckerMiddleware>() also on line 73

stephan-williamson commented on src/Epr.Reprocessor.Exporter.UI/Program.cs:
```diff
@@ -104,11 +109,7 @@
 //TODO: Dependency on enrollment and user account setup - Currently in progress
```
Can these TODOs be deleted? There is also one on line 101. Who's going to do the TODOs, just adding confusion in my opinion

stephan-williamson reviewed (commented):
Please include a description on your pull request to help persons looking at this in the future. Also no new unit tests, was this intentional? Thanks

venkat-eviden commented:
> Please include a description on your pull request to help persons looking at this in the future. Also no new unit tests, was this intentional? Thanks

I have updated the ticket title with the story requirement. 
I have updated the Unit tests now on the Home screen.

Ravi-SharmaGautam commented on src/Epr.Reprocessor.Exporter.UI/Extensions/ServiceProviderExtension.cs:
```diff
@@ -82,19 +81,21 @@ private static void ConfigureAuthorization(IServiceCollection services, IConfigu
     {
         services.AddScoped<ICookieService, CookieService>();
-        services.AddScoped<ISaveAndContinueService, SaveAndContinueService>();
-        services.AddScoped<ISessionManager<ReprocessorExporterRegistrationSession>, SessionManager<ReprocessorExporterRegistrationSession>>();
```
This has been removed. Can you check they are not by mistake?

Ravi-SharmaGautam commented on src/Epr.Reprocessor.Exporter.UI/Extensions/ServiceProviderExtension.cs:
```diff
@@ -108,15 +109,6 @@ private static void RegisterHttpClients(IServiceCollection services, IConfigurat
             client.BaseAddress = new Uri(facadeApiOptions.BaseEndpoint);
             client.Timeout = TimeSpan.FromSeconds(httpClientOptions.TimeoutSeconds);
         });
-
```
This too please check we don't want to remove registration for eprfacade

Ravi-SharmaGautam commented on src/Epr.Reprocessor.Exporter.UI/Program.cs:
```diff
@@ -48,6 +50,8 @@
     options.AllowedHosts = forwardedHeadersOptions.AllowedHosts;
 });
 
+services.Configure<HomeSettings>(builderConfig.GetSection("Links"));
```
Can you rename Homesettings to externalLinks there can be additional in future and they may not be specific to home?

Ravi-SharmaGautam commented on src/Epr.Reprocessor.Exporter.UI/ViewModels/HomeSettings.cs:
```diff
@@ -0,0 +1,13 @@
+﻿namespace Epr.Reprocessor.Exporter.UI.ViewModels
+{
+    public class HomeSettings
```
Can we rename to HomeViewModel rather than settings?

Ravi-SharmaGautam commented on src/Epr.Reprocessor.Exporter.UI/Views/Home/Index.cshtml:
```diff
@@ -1,23 +1,52 @@
-
+﻿@model Epr.Reprocessor.Exporter.UI.ViewModels.HomeSettings
 @{
     ViewData["Title"] = "Home";
```
All the non dynamic content should come from resource file as we need to support welsh translation. Check Stephen PRs on account creation repo for reference

stephan-williamson commented on src/Epr.Reprocessor.Exporter.UI/appsettings.json:
```diff
@@ -60,6 +60,13 @@
     "FindAndUpdateCompanyInformation": "https://find-and-update.company-information.service.gov.uk",
     "ProducerResponsibilityObligations": "#"
   },
+  "Links": {
```
Do we really need configurable endpoints, half the problems we have is messed up config? In other solutions they are constants
